Adds azd pipeline steps (#29)
Azure-Samples/azure-django-postgres-aca: lab/bonus-07-use-azd-template.md

# Bonus: Using AZD + GitHub Actions to Manage your Deployment

> You can also redeploy your project to Azure using GitHub Actions. This will allow you to redeploy your project to Azure when changes are made without manually running `AZD deploy`.

Reviewer: ```suggestion
You can also redeploy your project to Azure using GitHub Actions. This will allow GitHub to redeploy your project to Azure when changes are made without manually running `azd deploy`.
```

## Using GitHub Actions

> We've included a GitHub Action called [azure-dev.yaml](../.github/workflows/azure-dev.yaml). This action runs the provision and deployment steps using Federated Credentials.

Reviewer: ```suggestion
We've included a GitHub Action workflow file in [azure-dev.yaml](../.github/workflows/azure-dev.yaml). This workflow runs the provision and deployment steps using Federated Credentials.
```

To setup your GitHub repo, run the following command:

```shell
azd pipeline config
```

Enter the required information. 

https://user-images.githubusercontent.com/8632637/231233580-c0f1d520-f81d-4db5-931e-2b96cb7d2ec6.mp4

The information required by the workflow will be added to your GitHub repo.

![GitHub Actions Secrets and Variables](../assets/action-secrets-and-vars.png)

## Running the GitHub Action

> Once that is complete you can run the GitHub Action.

Reviewer: ```suggestion
Once that is complete you can run the GitHub Action workflow.
```

> We've intentionally commented out the automated deployment steps but you can uncomment them to automatically deploy your project to Azure. For now, we can manually trigger the deployment using the GitHub Actions tab.

Reviewer: Why's it commented out?
